src/Orchard.Web/Modules/Orchard.Widgets/Filters/WidgetFilter.cs

```diff
@@ -49,7 +49,7 @@ public class WidgetFilter : FilterProvider, IResultFilter {
                 return;
             }
 
-            var widgetParts = _widgetsService.GetWidgets(_layerEvaluationService.GetActiveLayerIds());
+            var widgetParts = _widgetsService.GetWidgetsWithEagerlyLoadedContainers(_layerEvaluationService.GetActiveLayerIds());
```

Comment:
This prevents n+1 problem caused by next loop that accesses commonPart.container for each widget

Reply:
In response to this comment from @sebastienros "I'd prefer a case by case improvement, where the issue is actually is. And also explaining what kind of data and scenario trigger the issues. Adding eager loading by default might be bad in other cases."
I did it here cause noone else consume this method, but I agree it could be used by others. I'm going to move the eagerly loading to the consumer code of this method.